annotation-web/src/main/java/eu/europeana/annotation/web/service/controller/jsonld/BaseJsonldRest.java

```diff
@@ -504,6 +513,50 @@ private boolean hasAdminRights(Authentication authentication) {
 	}
     }
 
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(String etag, HttpServletRequest request) throws HttpException {
+
+		String ifMatchHeader = request.getHeader(HttpHeaders.IF_MATCH);
+		if (ifMatchHeader != null) {
+		    try {
+				if (!etag.equals(ifMatchHeader))
+				    throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+					    ifMatchHeader);
+		    } catch (NumberFormatException e) {
+				throw new HeaderValidationException(I18nConstants.INVALID_PARAM_VALUE, HttpHeaders.IF_MATCH,
+					ifMatchHeader);
+		    }
+		}
+    }
+    
+    /**
+     * This method generates etag for response header.
+     * 
+     * @param modifiedDate The date of the last modification of annotaiton
+     * @return etag value
+     */
+    public String generateETag(Date modifiedDate) {
```

Comment:
Please move this method to api commons and add the format and version parameters

Reply:
moved to commons-api

Reply:
must remove the method from this class after refactoring